### PR TITLE
Remove shards and committees

### PR DESCRIPTION
**Pull request by terencechain**
Part of #3865 

This PR removes shards and crosslinks. Excuse the big diff, they are all due to the deletion of pbs, crosslink spec tests and all the crosslink transition function/test.

Change list:
* Modified `CommitteeCount` to `CommitteeCountAtSlot`
* Renamed crosslink committee to beacon committee
* Removed `AttestationDataSlot`
* Removed `ProcessCrosslinks`
* Removed `WinningCrosslink`
* Removed `CrosslinkDelta`
* Removed `ShardDelta`
* Removed `StartShard`
* Removed `attsForCrosslink`
* Removed `crosslinkDeltaPreCompute`
* Removed crosslink in state
* Removed crosslink spec tests
* Updated block transition `ProcessAttestation` to neglect crosslink
* Updated state transition to neglect crosslink
* Updated configs (12s in a slot, 32 slots per epoch, max validators per committee is 2048)

### CODE REVIEW

File: shared/params/config.go [hunk 1]
```diff
@@ -108,14 +108,14 @@ type DepositContractConfig struct {
...
-	BaseRewardsPerEpoch:      5,
+	BaseRewardsPerEpoch:      4,
 	DepositContractTreeDepth: 32,
 	SecondsPerDay:            86400,
 
 	// Misc constant.
-	ShardCount:                     1024,
 	TargetCommitteeSize:            128,
-	MaxValidatorsPerCommittee:      4096,
+	MaxValidatorsPerCommittee:      2048,
```
Comment (prestonvanloon) [inline shared/params/config.go]: Why is this different? This isn't mentioned in your description.

File: shared/params/config.go [hunk 2]
```diff
@@ -134,8 +134,8 @@ var defaultBeaconConfig = &BeaconChainConfig{
 
 	// Time parameter constants.
 	MinAttestationInclusionDelay:     1,
-	SecondsPerSlot:                   6,
-	SlotsPerEpoch:                    64,
+	SecondsPerSlot:                   12,
```
Comment (prestonvanloon) [inline shared/params/config.go]: Why?

File: beacon-chain/sync/validate_beacon_blocks_test.go [hunk 3]
```diff
@@ -98,7 +98,6 @@ func TestValidateBeaconBlockPubSub_BlockAlreadyPresentInCache(t *testing.T) {
 		t.Fatal(err)
 	}
 	msg := &ethpb.BeaconBlock{
-		Slot:       1,
```
Comment (prestonvanloon) [inline beacon-chain/sync/validate_beacon_blocks_test.go]: Why these changes in `beacon-chain/sync/validate_beacon_blocks_test.go`?

File: shared/params/config.go (hunk 1 above)
Comment (terencechain) [reply shared/params/config.go]: Updated descriptions

File: shared/params/config.go (hunk 2 above)
Comment (terencechain) [reply shared/params/config.go]: Updated descriptions, the rationale is more events will occur per slot in these modified proposals so upped 6 to 12

File: beacon-chain/sync/initial-sync/round_robin_test.go [hunk 4]
```diff
@@ -87,29 +87,29 @@ func TestRoundRobinSync(t *testing.T) {
 		},
 		{
 			name:               "Multiple peers with failures",
-			currentSlot:        320, // 5 epochs
-			expectedBlockSlots: makeSequence(1, 320),
+			currentSlot:        160, // 10 epochs
```
Comment (prestonvanloon) [inline beacon-chain/sync/initial-sync/round_robin_test.go]: Can you leave the number of slots the same, but just change the epochs? It looks like you reduced the blocks by half and also the finalized epoch by half in the peer data

File: beacon-chain/core/state/state_test.go [hunk 5]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 func TestGenesisBeaconState_OK(t *testing.T) {
-	if params.BeaconConfig().SlotsPerEpoch != 64 {
+	if params.BeaconConfig().SlotsPerEpoch != 32 {
 		t.Errorf("SlotsPerEpoch should be 64 for these tests to pass")
```
Comment (prestonvanloon) [inline beacon-chain/core/state/state_test.go]:
```suggestion
		t.Errorf("SlotsPerEpoch should be 32 for these tests to pass")
```
I think you can just delete this assertion. Seems like it had something to do with shard count

File: beacon-chain/core/helpers/validators_test.go [hunk 6]
```diff
@@ -144,7 +144,7 @@ func TestIsSlashableValidator_InactiveSlashed(t *testing.T) {
 func TestBeaconProposerIndex_OK(t *testing.T) {
 	ClearAllCaches()
 
-	if params.BeaconConfig().SlotsPerEpoch != 64 {
+	if params.BeaconConfig().SlotsPerEpoch != 32 {
 		t.Errorf("SlotsPerEpoch should be 64 for these tests to pass")
```
Comment (prestonvanloon) [inline beacon-chain/core/helpers/validators_test.go]: Delete these assertions if you could change this number and no tests fail. We had this because we wanted to fail faster on config changes

File: beacon-chain/sync/validate_beacon_blocks_test.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/sync/validate_beacon_blocks_test.go]: With slot 1, we get an error ` could not process slot from the future, slot time 1572483483 > current time 1572483471`.
Which makes sense because `genesis_time` is `time.Now` and we are passing in a block with slot 1 right away. This doesn't happen with 6s slot but only with 12s slot

File: beacon-chain/sync/initial-sync/round_robin_test.go (hunk 4 above)
Comment (terencechain) [reply beacon-chain/sync/initial-sync/round_robin_test.go]: Np. Since epoch duration is 32 slots. I'm going to use 320 blocks and just increased epoch to 8 (double from before)
